crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -294,18 +294,84 @@ impl Mesh {
         self.attributes.insert(name.into(), values);
     }
 
-    /// Retrieve the data currently set behind a vertex attribute.
+    /// Retrieves an immutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute(&self, name: impl Into<Cow<'static, str>>) -> Option<&VertexAttributeValues> {
         self.attributes.get(&name.into())
     }
 
+    /// Retrieves a mutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute_mut(
         &mut self,
         name: impl Into<Cow<'static, str>>,
     ) -> Option<&mut VertexAttributeValues> {
         self.attributes.get_mut(&name.into())
     }
 
+    /// Retrieves multiple mutable references to _distinct_ vertex attributes.
+    /// Returns None at index i if the mesh had no data for the vertex attribute given at index i.
+    ///
+    /// # Panics
+    /// Panics if any of the given attribute names resolve to the same attribute, as this would otherwise allow aliased mutability.
+    ///
+    /// # Example usage
+    /// ```
+    /// use crate::bevy_render::mesh::shape;
+    /// use crate::bevy_render::mesh::Mesh;
+    /// let mut mesh = Mesh::from(shape::Box::default());
+    /// let [positions, normals] = mesh.attribute_multi_mut([Mesh::ATTRIBUTE_POSITION, Mesh::ATTRIBUTE_NORMAL]);
+    /// ```
+    pub fn attribute_multi_mut<const N: usize>(
+        &mut self,
+        names: [impl Into<Cow<'static, str>>; N],
+    ) -> [Option<&mut VertexAttributeValues>; N] {
+        // SAFE: We verify that all the references returned are unique,
+        // therefore no aliased mutability is possible.
+        unsafe {
```

Comment:
Try to make your `unsafe` blocks as small as possible; it makes auditing functions like this easier to review and change.

Reply:
The unsafe block would be very small indeed for this function if I reduced it as much as possible - I _think_ there is only one line that strictly requires it. I personally think it makes sense to also enclose the surrounding code that ensures the safety of the "unsafe required" code.

But I can reduce the unsafe block's size if you insist :)

Reply:
Hmm, that's a sensible point. I think I would personally use a small unsafe block with a safety comments calling out how the rest of the function ensures that it's safe, but that's something I'm happy to punt on to more senior team members.

Reply:
I thought about it and I don't mind just that one line being the unsafe block :)